File: src/main/kotlin/com/stannis/dataModel/complexStatementTypes/ClassOrHeader.kt
package com.stannis.dataModel.complexStatementTypes

import com.stannis.dataModel.Statement
import com.stannis.dataModel.statementTypes.FunctionDeclarator
import com.stannis.dataModel.statementTypes.FunctionDefinition
import com.stannis.dataModel.statementTypes.LinkageSpecification
import com.stannis.dataModel.statementTypes.NameSpace

data class ClassOrHeader(
    override val `$type`: String = "ClassOrHeader",
    var directives: ArrayList<String>?,
    var globalDeclaration: ArrayList<DeclarationWithParent>?,
    var internDeclaration: ArrayList<DeclWithParent>?,
    var methodsWithFunctionCalls: ArrayList<FunctionDefinition>?,
    var functionCallsWithoutImplementation: ArrayList<FunctionDeclarator>?,
    var classList: ArrayList<ComplexCompositeTypeSpecifier>?,
    var linkageSpecification: ArrayList<LinkageSpecification>?,
    var namespaces: ArrayList<NameSpace>?,
    var externalMethods: ArrayList<String>?
    ) : Statement

File: Interpreter/Interpreter/Models/serialize/complexStatementTypes/ClassOrHeader.cs
using System.Collections.Generic;
using Interpreter.Models.serialize.statementTypes;

namespace Interpreter.Models.serialize.complexStatementTypes
{
        public class ClassOrHeader : IStatement
        {
                public IList<string> directives = new List<string>();

                public IList<DeclarationWithParent> globalDeclaration = new List<DeclarationWithParent>();

                public IList<DeclarationWithParent> internDeclaration = new List<DeclarationWithParent>();

                public IList<FunctionDefinition> methodsWithFunctionCalls = new List<FunctionDefinition>();

                public IList<FunctionDeclarator> functionCallsWithoutImplementation = new List<FunctionDeclarator>();

                public IList<ComplexCompositeTypeSpecifier> classList = new List<ComplexCompositeTypeSpecifier>(); 

                public IList<LinkageSpecification> linkageSpecification = new List<LinkageSpecification>(); 

                public IList<NameSpace> namespaces = new List<NameSpace>();

                public IList<string> externalMethods = new List<string>();
        }

}